sys/shell/commands/sc_gnrc_ipv6_nib.c

```diff
@@ -282,4 +294,74 @@ static int _nib_route(int argc, char **argv)
     return 0;
 }
 
+#if GNRC_IPV6_NIB_CONF_MULTIHOP_P6C
+static void _usage_nib_abr(char **argv)
+{
+#if GNRC_IPV6_NIB_CONF_6LBR
+    printf("usage: %s %s [show|add|del|help]\n", argv[0], argv[1]);
+    printf("       %s %s add <ipv6 global addr>\n",
+           argv[0], argv[1]);
+    printf("       %s %s del <ipv6 global addr>\n", argv[0], argv[1]);
+#else   /* GNRC_IPV6_NIB_CONF_6LBR */
+    printf("usage: %s %s [show|help]\n", argv[0], argv[1]);
+#endif  /* GNRC_IPV6_NIB_CONF_6LBR */
+    printf("       %s %s show\n", argv[0], argv[1]);
+}
+
+static int _nib_abr(int argc, char **argv)
+{
+    if ((argc == 2) || (strcmp(argv[2], "show") == 0)) {
+        gnrc_ipv6_nib_abr_t entry;
+        void *state = NULL;
+
+        while (gnrc_ipv6_nib_abr_iter(&state, &entry)) {
+            gnrc_ipv6_nib_abr_print(&entry);
+        }
+    }
+    else if ((argc > 2) && (strcmp(argv[2], "help") == 0)) {
+        _usage_nib_abr(argv);
+    }
+#if GNRC_IPV6_NIB_CONF_6LBR
+    else if ((argc > 3) && (strcmp(argv[2], "del") == 0)) {
+        ipv6_addr_t addr = IPV6_ADDR_UNSPECIFIED;
+
+        if (ipv6_addr_from_str(&addr, argv[3]) == NULL) {
+            _usage_nib_abr(argv);
+            return 1;
+        }
+        gnrc_ipv6_nib_abr_del(&addr);
+    }
+    else if ((argc > 3) && (strcmp(argv[2], "add") == 0)) {
+        gnrc_netif_t *netif;
+        ipv6_addr_t addr = IPV6_ADDR_UNSPECIFIED;
+
+        if (ipv6_addr_from_str(&addr, argv[3]) == NULL) {
+            _usage_nib_abr(argv);
+            return 1;
+        }
+        /* check addr */
+        if (ipv6_addr_is_link_local(&addr)) {
+            printf("address %s must be global\n", argv[3]);
+            return 1;
+        }
+        if (((netif = gnrc_netif_get_by_ipv6_addr(&addr)) == NULL) ||
+            gnrc_netif_is_6lbr(netif)) {
```

Comment:
```suggestion
            !gnrc_netif_is_6lbr(netif)) {
```
makes more sense.

Reply:
Does it? The messages context as logical formular would be "address assigned to interface netif" → "netif is 6LBR" which is exactly, what the if says.

Reply:
Yes but then you would not enter the "address %s is not assigned to a 6LBR interface" case afterwards.

Reply:
Made the formular (hopefully) more obvious.

Reply:
I think you just like double negations ;-)

Reply:
> I think you just like double negations ;-)

What you proposed was a double negation...

Reply:
I only see one negation now with my solution: `!/* negate */((netif = gnrc_netif_get_by_ipv6_addr(&addr)) != NULL) /* true if netif with addr exist */ && gnrc_netif_is_6lbr(netif) /* true if netif is 6LBR */`

Reply:
(remember in C `((netif = gnrc_netif_get_by_ipv6_addr(&addr)) != NULL)` and `(netif = gnrc_netif_get_by_ipv6_addr(&addr))` are the same. The first just makes it more clear that a pointer is checked.

Reply:
Yes I know. I just think it's easier to process mentally if it's if (condition A or nor condition B) than when it's if (not (condition A and condition B)).

But that's your module, so if you like it better that way I won't object - feel free to squash. :)